src/KM_FormMain.pas

```diff
@@ -436,12 +436,21 @@ procedure TFormMain.RGPlayerClick(Sender: TObject);
 
 
 procedure TFormMain.chkSuperSpeedClick(Sender: TObject);
+var Speed: Single;
 begin
   if (gGameApp.Game = nil)
-  or (gGameApp.Game.IsMultiplayer and not MULTIPLAYER_SPEEDUP and not gGameApp.Game.IsReplay) then
+  or (gGameApp.Game.IsMultiplayer
+    and not gGameApp.Game.IsMPGameSpeedUpAllowed
+    and not MULTIPLAYER_SPEEDUP
+    and not gGameApp.Game.IsReplay) then
     Exit;
 
-  gGameApp.Game.SetGameSpeed(IfThen(chkSuperSpeed.Checked, 300, 1), False);
+  if gGameApp.Game.IsMPGameSpeedUpAllowed then
```

Comment:
Default MP speed depends of game options and time (before/after) PT, while Default SP speed is always 1. Here we set normal speed to 1 or to gGameApp.Game.GetDefaultMPGameSpeed.
May be we can check gGameApp.Game.IsMultiplayer. Or better add new method which returns Default speed (for sp - 1, for mp - GetDefaultMPGameSpeed)